src/relax/op/op_common.h

```diff
@@ -127,30 +115,21 @@ inline StructInfo InferStructInfoUnary(const Call& call, const BlockBuilder& ctx
         << " requires the input tensor to have float dtype. However, the given input dtype is "
         << input_sinfo->dtype);
   }
-  DataType output_dtype = f_compute_out_dtype(call, ctx, input_sinfo);
-  if (input_sinfo->shape.defined()) {
-    return TensorStructInfo(input_sinfo->shape.value(), output_dtype);
-  } else {
-    return TensorStructInfo(output_dtype, input_sinfo->ndim);
-  }
+  auto output_sinfo = make_object<TensorStructInfoNode>(*input_sinfo.get());
+  output_sinfo->dtype = f_compute_out_dtype(input_sinfo);
+  return TensorStructInfo(output_sinfo);
 }
 
 template <bool require_float_dtype>
 StructInfo InferStructInfoUnaryArith(const Call& call, const BlockBuilder& ctx) {
   return InferStructInfoUnary<require_float_dtype>(
-      call, ctx,
-      [](const Call& call, const BlockBuilder& ctx, const TensorStructInfo& input_sinfo) {
-        return input_sinfo->dtype;
-      });
+      call, ctx, [](const TensorStructInfo& input_sinfo) { return input_sinfo->dtype; });
 }
 
 template <bool require_float_dtype>
 StructInfo InferStructInfoUnaryCheck(const Call& call, const BlockBuilder& ctx) {
   return InferStructInfoUnary<require_float_dtype>(
-      call, ctx,
-      [](const Call& call, const BlockBuilder& ctx, const TensorStructInfo& input_sinfo) {
-        return DataType::Bool();
-      });
+      call, ctx, [](const TensorStructInfo& input_sinfo) { return DataType::Bool(); });
```

Comment:
Last one: here we can directly pass `/*require_float_dtype=*/false` I think.

Reply:
On line 131.

Reply:
As the template parameter, should we write as `InferStructInfoUnary</*require_float_dtype=*/false>` or directly 
`InferStructInfoUnary<false>`?

Reply:
false is okay I think 🤦

Reply:
Ok since now it is not a template function, I will put its implementation in `op_common.cc`